de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlDeliveryOrderRepository.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * #%L
+ * de.metas.shipper.gateway.dhl
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.shipper.gateway.dhl;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import de.metas.shipper.gateway.dhl.model.DhlCustomDeliveryData;
+import de.metas.shipper.gateway.dhl.model.DhlServiceType;
+import de.metas.shipper.gateway.dhl.model.I_DHL_ShipmentOrder;
+import de.metas.shipper.gateway.dhl.model.I_DHL_ShipmentOrderRequest;
+import de.metas.shipper.gateway.spi.DeliveryOrderId;
+import de.metas.shipper.gateway.spi.DeliveryOrderRepository;
+import de.metas.shipper.gateway.spi.model.Address;
+import de.metas.shipper.gateway.spi.model.ContactPerson;
+import de.metas.shipper.gateway.spi.model.CountryCode;
+import de.metas.shipper.gateway.spi.model.DeliveryOrder;
+import de.metas.shipper.gateway.spi.model.DeliveryPosition;
+import de.metas.shipper.gateway.spi.model.PackageDimensions;
+import de.metas.shipper.gateway.spi.model.PickupDate;
+import de.metas.util.Check;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.lang.ITableRecordReference;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Repository;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+@Repository
+public class DhlDeliveryOrderRepository implements DeliveryOrderRepository
+{
+	private static final Logger logger = LoggerFactory.getLogger(DhlDeliveryOrderRepository.class);
+
+	@Override
+	public String getShipperGatewayId()
+	{
+		return DhlConstants.SHIPPER_GATEWAY_ID;
+	}
+
+	@Override
+	public ITableRecordReference toTableRecordReference(final DeliveryOrder deliveryOrder)
```

Comment:
Please add lombok's `@NonNull` wherever you assume the  paramter to be not-null. Makes for more informative NPEs!
For the other parameters, pls prepend javax's `@Nullable`

Reply:
don't worry, i'm adding annotations as much as possible coz intellij likes having them as well!

Reply:
ok...but here you don't. why? forgotten?